cdap-proto/src/main/java/co/cask/cdap/proto/metadata/lineage/FieldLineageSummaryRecord.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.proto.metadata.lineage;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Class representing the individual record in the {@link FieldLineageSummary}.
```

Comment:
We could name this "DatasetWithFields" or something similar. It would be more descriptive. Or is the idea that this class can be extended with additional fields in the future?

Reply:
Also saying "class representing a record" is kind of moot point. Doesn't each record represent a dataset and a subset of its fields, where that dataset is either an origin or a destination in the lineage graph?

Reply:
Renamed the class to `DatasetField` and updated the javadoc.